generic_scripts/vchamber_temp_excursion_induced_by_electrons.py

```diff
@@ -0,0 +1,192 @@
+    _dose_al = 2.001
+    SPC = {}
+    for k, v in _DOSES.items():
+        SPC[k] = v / _dose_al * _spc_al
+
+    SPEC_HEAT = {  # [J/mole/K]
+        'Graphite': 8.52,
+        'Aluminum': 24.2,
+        'Titaniun': 25.06,
+        'Copper': 24.44,
+        'Tungsten': 24.27,
+    }
+    ATOMIC_WEIGHT = {  # [g/mole]
+        'Graphite': 12.01,
+        'Aluminum': 26.98,
+        'Titaniun': 47.87,  # for Ti Only
+        'Copper': 63.55,
+        'Tungsten': 183.84,
+    }
+    FUSION_TEMP = {  # [K]
+        'Graphite': 3915,  # sublimation
+        'Aluminum': 933,
+        'Titaniun': 1900,  # 1950
+        'Copper': 1358,
+        'Tungsten': 3695,
+    }
+
+    MATERIALS = ('Aluminum', 'Titaniun', 'Copper', 'Graphite', 'Tungsten')
+
+    def __init__(self, model, material='Al', current=0.350):
+        """Create object.
+
+        Args:
+            model (pyaccel.accelerator.Accelerator): model of the ring.
+            material (str, optional): material of the vacuum chamber.
+                Defaults to 'Al'.
+            current (float, optional): Current of the electron beam in [A].
+                Defaults to 0.350.
+
+        """
+        self.current = current
+        self.model = model
+        self._material = self.MATERIALS[0]
+        self.material = material
+
+        self.eqpar = None
+        self.energy_density = None
+        self.peak_dose = None
+        self.temp_excursion = None
+        self.update()
+
+    @property
+    def material(self):
+        """Material of the vacuum chamber.
+
+        Returns:
+            str: material.
+
+        """
+        return self._material
+
+    @material.setter
+    def material(self, val):
+        if isinstance(val, (int, np.int_)) and 0 <= val < len(self.MATERIALS):
+            self._material = self.MATERIALS[val]
+        elif isinstance(val, str):
+            idc = [m for m in self.MATERIALS if val.lower() in m.lower()]
+            if not idc:
+                raise ValueError('Wrong value for material')
+            self._material = idc[0]
+        else:
+            raise ValueError('Wrong value for material')
+
+    @property
+    def melting_temp(self):
+        """Melting temperature of the vacuum chamber material in [K].
+
+        Returns:
+            float: melting temperature.
+        """
+        return self.FUSION_TEMP[self._material]
+
+    def update(self):
+        """Update relevant quantities."""
+        mod = self.model
+        harm = mod.harmonic_number
+        energy = mod.energy  # [eV]
+        f_rf = pa.optics.get_rf_frequency(mod)
+        f_rev = f_rf / harm  # [Hz]
+        charge = self.current / f_rev  # [C]
+
+        self.eqpar = pa.optics.EqParamsFromBeamEnvelope(mod)
+        self.spos = np.array(pa.lattice.find_spos(mod, indices='closed'))
+        sigx = self.eqpar.sigma_rx
+        sigy = self.eqpar.sigma_ry
+
+        # energy density in [J/m^2]
+        self.energy_density = charge * energy / 2 / np.pi / sigx / sigy
```

Comment:
maybe the class interface should make explicit that various calculated properties are for normal beam collision, and not grazing collisions.

Reply:
I'm not sure if that is True. The particles will penetrate the chamber and the cross-section area of the cylinder formed along the depth of the material is the same for both types of collision. Since this is an open question, I prefer not to add any comment.

Reply:
we talked offline and agreed not to add any comment.